core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -538,10 +538,37 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     }
   }
 
+  test("Killing tasks") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+
+    SparkContextSuite.isTaskStarted = false
+    SparkContextSuite.taskKilled = false
+
+    val listener = new SparkListener {
+      override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+        eventually(timeout(10.seconds)) {
+          assert(SparkContextSuite.isTaskStarted)
+        }
+        if (!SparkContextSuite.taskKilled) {
+          SparkContextSuite.taskKilled = true
+          sc.killTask(taskStart.taskInfo.taskId, "first attempt will hang")
+        }
+      }
+    }
+    sc.addSparkListener(listener)
+    sc.parallelize(1 to 1).foreach { x =>
```

Comment:
Can you wrap this part in an eventually statement so that the test will fail relatively quickly (i.e., in less than the 250m overall test timeout) if the kill fails